src/main/java/org/xerial/snappy/SnappyOutputStream.java

```diff
@@ -365,15 +365,21 @@ protected void dumpOutput()
     protected void compressInput()
             throws IOException
     {
-        if (inputCursor <= 0) {
-            return; // no need to dump
-        }
-
+        //  generate header
         if (!headerWritten) {
             outputCursor = writeHeader();
             headerWritten = true;
         }
 
+        if (inputCursor <= 0) {
+            return; // no need to dump
+        }
+
+//        if (!headerWritten) {
```

Comment:
Please remove these comments as we already have the same code block in the above